app/src/ui-blokada/kotlin/core/AddDnsActivity.kt

```diff
@@ -86,9 +88,32 @@ class AddDnsActivity : Activity() {
             stepView.next()
         })
 
+        val dotHostVB = EnterDotHostVB(ktx, accepted = {
+            if (it.isEmpty()) {
+                dotEnabled = false
+            } else {
+                dotEnabled = true
+                var host = it
+                var portString = "853"
+                if (it.contains(":")) {
+                    val parts = it.split(":")
+                    host = parts[0]
+                    portString = parts[1]
+                }
+                var port = portString.toInt()
+                for (i in servers.indices) {
+                    if (servers[i] != null) {
+                        servers[i] = InetSocketAddress(InetAddress.getByAddress(host, servers[i]!!.address.address), port)
```

Comment:
This logic could probably be better. If the user specifies a DoT hostname, we iterate over the previously specified servers, changing them to InetSocketAddresses with the DoT hostname as the hostname and the previously entered IP address as the IP address. We then set the `dotEnabled` flag. It's a bit awkward.

Reply:
The best option might be to rewrite the first input field to also allow a valid hostname instead of a IPv4 address and to change the rest of the steps accordingly. Validation of the hostname would be nice either way,

Reply:
Added hostname validation. 

I don't think having a hostname instead of an IP address makes sense, though. The IP address is what's used for the DNS lookup, the hostname is just for verifying the cert. See this test, where both the IP address and hostname are specified: https://getdnsapi.net/query/

Reply:
I should have been a bit more clear. There would be 2 possibilities:
User enters a IP in the first field-> User is prompted to input another IP for normal DNS 
User enters a domain in the first field-> User is prompted to input the IP

But I don't know if this would be confusing,

Reply:
How about adding a new button to the DNS page, in addition to the "Add new DNS..." button, "Add new DNS-over-TLS..."? If the user selects the DNS button, the flow is the same as before these changes:
```
IP 1 -> IP 2 -> Display Name
```
and if they select the DoT button, the flow is:
```
IP 1 -> IP 2 -> DoT Hostname -> Display Name
```
For each flow, all fields would be required.

Reply:
Looks like adding a branching flow would involve rewriting VBStepView, since currently it only supports a single list of ViewBinders. The simplest options would be to have an optional DoT hostname field (like is in the PR now), or to have two different flows, one for standard DNS and one for DoT. The DoT flow can be simplified to:
```
DoT Hostname -> Display Name
```
by using getAllByName, so that could simplify things if it has its own flow.

Reply:
Per above comment, created a separate flow, with a new button, for creating a DNS over TLS server. The change is in commit 0eeaa9a.

Reply:
I think this is the best solution. :+1: